src/polynomials/standard-basis/immutable-polynomial.jl

```diff
@@ -142,8 +142,11 @@ end
 
 # special cases are much more performant
 derivative(p::ImmutableDensePolynomial{B,T,X,0}) where {B<:StandardBasis,T,X} = p
+derivative(p::ImmutableDensePolynomial{B,T,X,1}) where {B<:StandardBasis,T,X} = zero(p)
+
 function derivative(p::ImmutableDensePolynomial{B,T,X,N}) where {B<:StandardBasis,T,X,N}
     N == 0 && return p
+    N == 1 && return zero(p)
```

Comment:
It seems like rather than add a case, we can remove the `N==0` case which is captured in the special case above.

Reply:
I think, both cases (`N == 0` and `N == 1`) are covered by the more specific functions in lines 145 and 146, so could be both removed.